host/sampi/http.go

```diff
@@ -95,23 +109,30 @@ func startJSONEventStreaming(w http.ResponseWriter) *json.Encoder {
 
 // HTTP Route Handles
 func (c *HTTPAPI) ListHosts(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	l := c.logger.New("fn", "ListHosts")
 	ret, err := c.Cluster.ListHosts()
 	if err != nil {
+		l.Error("error occured", "err", err)
 		httphelper.Error(w, err)
 		return
 	}
+	l.Debug("writing http response")
 	httphelper.JSON(w, 200, ret)
 }
 
 func (c *HTTPAPI) RegisterHost(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	l := c.logger.New("fn", "RegisterHost")
 	h := &host.Host{}
 	if err := json.NewDecoder(r.Body).Decode(&h); err != nil {
+		l.Error("json decoding failed", "err", err)
 		httphelper.Error(w, err)
 		return
 	}
 
 	if h.ID == "" {
-		httphelper.Error(w, errors.New("sampi: host id must not be blank"))
+		err := errors.New("sampi: host id must not be blank")
+		l.Error("host id is blank", "err", err)
```

Comment:
No need for the `err` key, I would just change the message to something like `error registering host, the given host ID is blank`